lib/src/parse/stylesheet.dart

```diff
@@ -2894,33 +2895,70 @@ abstract class StylesheetParser extends Parser {
     assert(scanner.peekChar() == $lparen);
     switch (name) {
       case "calc":
+      case "sqrt":
+      case "sin":
+      case "cos":
+      case "tan":
+      case "asin":
+      case "acos":
+      case "atan":
+      case "exp":
+      case "sign":
         var arguments = _calculationArguments(1);
         return CalculationExpression(name, arguments, scanner.spanFrom(start));
 
+      case "abs":
+        return _tryArgumentsCalculation(name, start, 1);
+
+      case "hypot":
+        var arguments = _calculationArguments();
+        return CalculationExpression(name, arguments, scanner.spanFrom(start));
+
       case "min":
       case "max":
         // min() and max() are parsed as calculations if possible, and otherwise
         // are parsed as normal Sass functions.
-        var beforeArguments = scanner.state;
-        List<Expression> arguments;
-        try {
-          arguments = _calculationArguments();
-        } on FormatException catch (_) {
-          scanner.state = beforeArguments;
-          return null;
-        }
-
+        return _tryArgumentsCalculation(name, start, null);
+
+      case "pow":
+      case "log":
+      case "atan2":
+      case "mod":
+      case "rem":
+        var arguments = _calculationArguments(2);
         return CalculationExpression(name, arguments, scanner.spanFrom(start));
 
       case "clamp":
         var arguments = _calculationArguments(3);
         return CalculationExpression(name, arguments, scanner.spanFrom(start));
 
+      case "round":
+        return _tryArgumentsCalculation(name, start, 3);
+
       default:
         return null;
     }
   }
 
+  // Global calculation functions should allow named arguments
+  // for backwards-compatibility. If the parameter contains the ':' symbol
+  // the parser takes it as a separator.
+  // For instance, abs($number: 3) is taken as two arguments.
+  //
+  // This function returns the calculation as it was inputted
+  // if it contains more parameters than expected.
```

Comment:
This is misleading. If there are more than `maxArgs` available, `_calculationArguments()` will throw an error and this will return `null`, it won't return the `CalculationExpression`.